CodingTracker.DreamFXX/DatabaseManager.cs

```diff
@@ -0,0 +1,138 @@
+            using (var connection = new SqliteConnection(_connectionString))
+            {
+                connection.Open();
+
+                connection.Execute(@"DELETE FROM MyCodingTracker
+                                            WHERE Date = @date", new { date });
+            }
+        }
+
+        public void UpdateRecord(string? oldDate, string? newDate, string? startTime, string? endTime, string? duration)
+        {
+            using (var connection = new SqliteConnection(_connectionString))
+            {
+                connection.Open();
+
+                if (@startTime is null && @endTime is null)
+                {
+                    connection.Execute(@"UPDATE MyCodingTracker
+                                             SET Date = @newDate
+                                             WHERE Date = @oldDate", new {newDate,oldDate});
+                }
+                else if (@newDate is null)
+                {
+                    connection.Execute(@"UPDATE MyCodingTracker
+                                             SET StartTime = @startTime, EndTime = @endTime, Duration = @duration
+                                             WHERE Date = @oldDate", new {startTime, endTime, duration, oldDate});
+                }
+                else
+                {
+                    connection.Execute(@"UPDATE MyCodingTracker
+                                              SET Date = @newDate, StartTime = @startTime, EndTime = @endTime, Duration = @duration
+                                              WHERE Date = @oldDate", new {newDate, startTime, endTime, duration, oldDate});
+                }
+            }
+        }
+
+        public List<CodingSession> ReadFromDb()
+        {
+            List<CodingSession> codingSessions = new();
+
+            var connection = new SqliteConnection(_connectionString);
+            connection.Open();
+
+            var sqlQuery = "SELECT * FROM MyCodingTracker";
+
+            using (var command = new SqliteCommand(sqlQuery, connection)) 
+            using (var reader = command.ExecuteReader())
+
+            {
+                if (reader.HasRows)
+                    while (reader.Read())
+                    {
+                        var model = new CodingSession
+                        {
+                            Id = Convert.ToInt32(reader["Id"]),
+                            Date = reader["Date"].ToString()!,
+                            StartTime = reader["StartTime"].ToString()!,
+                            EndTime = reader["EndTime"].ToString()!,
+                            Duration = reader["Duration"].ToString()!
+                        };
+
+                        codingSessions.Add(model);
+                    }
+                else
+                    AnsiConsole.MarkupLine("[red]No records found in the database.[/]");
+            }
+            return codingSessions;
+        }
+
+        public CodingSession? ReadSingleRecord(string? date)
```

Comment:
🔴 **This is still not resolved**

Note: Dapper vs ADO.NET

![image](https://github.com/user-attachments/assets/13ce2626-33aa-4456-a8c2-d6c8da725d8d)